### PR TITLE
Group based access

### PR DESCRIPTION
**Pull request by u20612894**
Schema has been updated with group access rules in addition to owner based access. This means that from now on only user pools can be used for authentication not API keys

### CODE REVIEW

**Review by Tymac15 — COMMENTED** [review]
Code seems good but check the secrets that were detected?

**Comment by u20612894** [comment]
> Code seems good but check the secrets that were detected?

I branched of main to bring the cleanup into backend. That's where the secrets are coming from

**Comment by Bandisa** [comment]
This merging into backend while it is mostly frontend changes.

**Comment by u20612894** [comment]
> This merging into backend while it is mostly frontend changes.

Its bringing in changes from main

**Comment by Bandisa** [comment]
> > This merging into backend while it is mostly frontend changes.
> 
> 
> 
> Its bringing in changes from main

Okay that makes sense.

File: amplify/backend/test/unit/function/pronto5f713f59PostAuthentication/add-to-group.test.js
Comment (Bandisa) [inline amplify/backend/test/unit/function/pronto5f713f59PostAuthentication/add-to-group.test.js:1]: We need to restore these changes prior before we merge this PR.

**Review by ghost — DISMISSED** [review]

File: amplify/backend/test/unit/function/pronto5f713f59PostAuthentication/add-to-group.test.js
Comment (Bandisa) [reply amplify/backend/test/unit/function/pronto5f713f59PostAuthentication/add-to-group.test.js:1]: I will do this tomorrow.

Comment (Bandisa) [reply amplify/backend/test/unit/function/pronto5f713f59PostAuthentication/add-to-group.test.js:1]: working on this.

Comment (Bandisa) [reply amplify/backend/test/unit/function/pronto5f713f59PostAuthentication/add-to-group.test.js:1]: # Update
this has been resolved, more on #322

**Review by Bandisa — APPROVED** [review]
Looks great. I have also restored the previously deleted tests.

**Review by ShashinGounden — APPROVED** [review]

**Review by Tymac15 — APPROVED** [review]

**Pull request merged**